MH-13318, Update to Apache Karaf 4.2.2

lkiesow opened the pull request:
This patch updates Apache Karaf to its latest stable version.

lkiesow committed:
MH-13318, Update to Apache Karaf 4.2.2

This patch updates Apache Karaf to its latest stable version.

JamesUoM commented:
Karaf 4.2.3 was released yesterday (11th Feb) if we want to be bleeding edge.

JamesUoM commented on pom.xml:
```diff
@@ -35,14 +35,14 @@
     <osgi.core.version>5.0.0</osgi.core.version>
     <osgi.enterprise.version>5.0.0</osgi.enterprise.version>
-    <pax.web.version>7.0.0</pax.web.version>
+    <pax.web.version>7.2.3</pax.web.version>
```
The news article (and pom) for the Karaf 4.2.2 release list pax web v7.2.5 (though confusingly the release note mention v7.2.4).

pax web 7.2.3 was supported by karaf 4.2.1

lkiesow replied:
Not sure where I did take the version from. The notes about the versions are always kind of messy :-(
I'll look into it and fix it.

JamesUoM replied:
Do we need to specify it at all if Karaf is pulling it in already?

lkiesow replied:
> Do we need to specify it at all if Karaf is pulling it in already?

Good question and it seems like we don't need it. The specified version was used at a few more places but cleaning that out results in a fully functional system so I updated this pull request accordingly.

lkiesow committed:
MH-13318, Let Karaf specify the Pax-Web version

This patch removes the declaration of a specific Pax-Web version as
Opencast's dependency since Karaf ships it anyway and we should be able
to rely on that instead.

lkiesow commented:
> Karaf 4.2.3 was released yesterday (11th Feb) if we want to be bleeding edge.

I did take a short look at it but it seems like the included eclipselink update breaks the OAI-PMH tests. While it could very well be the fault of those tests, I'm leaving it at 4.2.2 for now. I might take a look at it again in a few weeks if things here are a bit quieter and I have more time on my hands.
